src/eckit/io/PooledFile.cc

```diff
@@ -21,12 +21,31 @@
 #include "eckit/io/PooledFile.h"
 #include "eckit/log/Bytes.h"
 
-
 namespace eckit {
-
 class PoolFileEntry;
+}
+
+namespace {
+
+class Pool {
+
+private:
+    Pool() {}
+    std::map<eckit::PathName, std::unique_ptr<eckit::PoolFileEntry>> filePool_;
+    std::mutex filePoolMutex_;
```

Comment:
You just have to be careful not missing threading models, std::mutex and eckit::Mutex (a recursive mutex based on pthread_mutex_t)